packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-table.tsx

```diff
@@ -0,0 +1,251 @@
+  connectionStringUrl: ConnectionStringUrl;
+}): React.ReactElement {
+  const [options, setOptions] = React.useState<Partial<UrlOption>[]>([]);
+  const [errorMessge, setErrorMessge] = React.useState('');
+
+  useEffect(() => {
+    const options: Partial<UrlOption>[] = getUrlOptions(connectionStringUrl);
+    if (!options.length) {
+      options.push({ name: undefined, value: undefined });
+    }
+    setOptions(options);
+  }, [connectionStringUrl]);
+
+  const addUrlOption = () => {
+    setErrorMessge('');
+    // Use case: User clicks on `Add url option` button and then clicked again without completing existing entry.
+    // Don't add another option in such case
+    if (options.find(({ name }) => !name)) {
+      setErrorMessge('Please complete existing option.');
+      return;
+    }
+    const newOptions = [...options, { name: undefined, value: undefined }];
+    setOptions(newOptions);
+  };
+
+  const updateUrlOption = (
+    currentName?: UrlOption['name'],
+    name?: UrlOption['name'],
+    value?: string
+  ) => {
+    setErrorMessge('');
+    const indexOfUpdatedOption = options.findIndex(
+      (option) => option.name === currentName
+    );
+    const option = {
+      name,
+      value,
+    };
+    const newOptions = [...options];
+    newOptions[indexOfUpdatedOption] = option;
+    setOptions(newOptions);
+
+    if (name) {
+      updateConnectionFormField({
+        type: 'update-search-param',
+        // If the user is selecting key for the first time, currentName is undefined and we choose the selected value.
+        currentKey: currentName ?? name,
+        // If the keys are same, then user is changing value. So we set newKey to undefined.
+        newKey: currentName === name ? undefined : name,
+        value: value ?? '',
+      });
+    }
+  };
+
+  const deleteUrlOption = (name?: UrlOption['name']) => {
+    setErrorMessge('');
+    const indexOfDeletedOption = options.findIndex(
+      (option) => option.name === name
+    );
+    const newOptions = [...options];
+    newOptions.splice(indexOfDeletedOption, 1);
+    setOptions(newOptions);
+    if (name) {
+      updateConnectionFormField({
+        type: 'delete-search-param',
+        key: name,
+      });
+    }
+  };
+
+  return (
+    <>
+      <Table
+        data-testid="url-options-table"
+        data={options}
+        columns={[
+          <TableHeader
+            key={'name'}
+            label="Key"
+            className={tableHeaderStyles}
+          />,
+          <TableHeader
+            key={'value'}
+            label="Value"
+            className={tableHeaderStyles}
+          />,
+        ]}
+      >
+        {({ datum }: { datum: Partial<UrlOption> }) => (
+          <Row
+            key={datum.name}
+            data-testid={
+              datum.name ? `${datum.name}-table-row` : 'new-option-table-row'
+            }
+          >
+            <Cell>
+              <Select
+                placeholder="Select key"
```

Comment:
Probably not the best solution, but something like this seems to do the trick for the initial width styling
```ts
const selectStyles = css({
  width: 300
});
```

```suggestion
                placeholder="Select key"
                className={selectStyles}
```

<img width="558" alt="Screen Shot 2022-01-06 at 12 56 09 PM" src="https://user-images.githubusercontent.com/1791149/148428624-42a25a51-ce87-404d-a02c-00435209b603.png">

Reply:
using absolute number works, but yeah not a clean way. will use this approach.